### PR TITLE
SNOW-270946 Progress percentage computation to handle the case when f…

### PR DESCRIPTION
**Pull request by sfc-gh-cshi**
…ile size is zero

### CODE REVIEW

**Review by sfc-gh-kwagner — APPROVED** [review]

File: test/unit/test_put_get.py [hunk 1]
```diff
@@ -74,3 +78,16 @@ def test_put_error(tmpdir):
...
+    """Tests for raise_put_get_error flag (now turned on by default) in SnowflakeFileTransferAgent."""
+    tmp_dir = str(tmpdir.mkdir('putfiledir'))
+    file_name = path.join(tmp_dir, 'zero_file1')
+    from pathlib import Path
+    Path(file_name).touch()
+    func_callback = SnowflakeS3ProgressPercentage(file_name, 0)
+    func_callback(1)
+
+    func_callback = SnowflakeAzureProgressPercentage(file_name, 0)
+    func_callback(1)
```
Comment (sfc-gh-mkeller) [inline test/unit/test_put_get.py:93]: Check out [tmp_path fixture](https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture), it'll cut out a bunch of your code. Also please be aware of our existing type hints. The percentage classes expect a `str` path.
```suggestion
def test_percentage(tmp_path):
    """Tests for raise_put_get_error flag (now turned on by default) in SnowflakeFileTransferAgent."""
    file = tmp_path / 'zero_file1'
    file.touch()
    func_callback = SnowflakeS3ProgressPercentage(str(file), 0)
    func_callback(1)

    func_callback = SnowflakeAzureProgressPercentage(str(file), 0)
    func_callback(1)
```

File: src/snowflake/connector/file_transfer_agent.py [hunk 2]
```diff
@@ -107,6 +107,11 @@ def _update_progress(
     return progress == 1.0
 
 
+def percent(seen_so_far, size):
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/file_transfer_agent.py:110]: Please verify these
```suggestion
def percent(seen_so_far:int , size: float) -> float:
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: test/unit/test_put_get.py [hunk 3]
```diff
@@ -74,3 +79,19 @@ def test_put_error(tmpdir):
...
+def test_percentage(tmpdir):
+    """Tests for ProgressPercentage classes."""
+    assert 1.0 == percent(0, 0)
+    assert 1.0 == percent(20, 0)
+    assert 1.0 == percent(40, 20)
+    assert 0.5 == percent(14, 28)
+
+    file_path = tmpdir / 'zero_file1'
+    from pathlib import Path
+    Path(file_path).touch()
```
Comment (sfc-gh-mkeller) [inline test/unit/test_put_get.py:93]: Please take a look at https://docs.pytest.org/en/stable/tmpdir.html `tmpdir` and `tmp_path` are not the same.
```suggestion
def test_percentage(tmp_path):
    """Tests for ProgressPercentage classes."""
    assert 1.0 == percent(0, 0)
    assert 1.0 == percent(20, 0)
    assert 1.0 == percent(40, 20)
    assert 0.5 == percent(14, 28)

    file_path = tmpdir / 'zero_file1'
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: src/snowflake/connector/file_transfer_agent.py [hunk 4]
```diff
@@ -208,7 +213,7 @@ def __call__(self, bytes_amount: int):
         with self._lock:
             if self._output_stream:
                 self._seen_so_far += bytes_amount
-                percentage = float(self._seen_so_far / self._size)
+                percentage = percent(self._seen_so_far, self._size)
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/file_transfer_agent.py:216]: `self._size` is initialized as `float(filesize)` so you are not respecting the type hints of `percent`

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
🚢 🚢

**Pull request merged**